src/lib/bootloader/systemdboot.rb

```diff
@@ -0,0 +1,304 @@
+    def create_menue_entries
+      cmdline_file = File.join(Yast::Installation.destdir, "/etc/kernel/cmdline")
+      if Yast::Stage.initial
+        # kernel-install script needs the "root=<device>" entry in kernel parameters.
+        File.open(cmdline_file, "w+") do |fw|
+          fw.puts("root=#{Yast::BootStorage.root_partitions.first.name}")
+        end
+      end
+
+      Dir.foreach(File.join(Yast::Installation.destdir, "/usr/lib/modules")) do |kernel_name|
+        next if [".", ".."].include?(kernel_name)
+
+        kernel_file = File.join("/usr/lib/modules", kernel_name, "vmlinuz")
+        if File.exist?(File.join(Yast::Installation.destdir, kernel_file))
+          begin
+            Yast::Execute.on_target!(KERNELINSTALL, "add",
+              File.basename(kernel_name), kernel_file)
+          rescue Cheetah::ExecutionFailed => e
+            Yast::Report.Error(
+            format(_(
+                     "Cannot create systemd-boot menue entry:\n" \
+                     "Command `%{command}`.\n" \
+                     "Error output: %{stderr}"
+                   ), command: e.commands.inspect, stderr: e.stderr)
+          )
+            break
+          end
+        end
+      end
+      File.delete(cmdline_file) if Yast::Stage.initial
+    end
+
+    def read_menue_timeout
+      filename = File.join(Yast::Installation.destdir, LOADERCONF)
+      return unless File.file?(filename)
+
+      f = File.open(filename, "r")
+      f.each_line do |line|
+        line.lstrip!
+        next if line.start_with?("#")
+
+        if line.start_with?("timeout")
+          self.menue_timeout = line.split[1].to_i
+          break
+        end
+      end
+    end
+
+    def write_menue_timeout
+      filename = File.join(Yast::Installation.destdir, LOADERCONF)
+      return unless File.file?(filename)
+
+      output = []
+      entry_set = false
+      f = File.open(filename, "r")
+      f.each_line do |line|
+        line.lstrip!
+        if line.start_with?("timeout") || line.start_with?("#timeout")
+          output << "timeout #{menue_timeout}\n"
+          entry_set = true
+          next
+        end
+        output << line
+      end
+      output << "timeout #{menue_timeout}" unless entry_set
+      File.open(filename, "w+") do |fw|
+        fw.puts(output)
+      end
+    end
+
+    def bootloader_is_installed
+      Yast::Execute.on_target(BOOTCTL, "is-installed", allowed_exitstatus: 1) == 0
+    end
+
+    def remove_secure_boot_settings
+      del_files = ["/boot/efi/EFI/systemd/grub.efi",
+                   "/boot/efi/EFI/systemd/systemd-bootx64.efi",
+                   "/boot/efi/EFI/systemd/MokManager.efi"]
+      del_files.each do |f|
+        filename = File.join(Yast::Installation.destdir, f)
+        File.delete(filename) if File.exist?(filename)
+      end
+    end
+
+    def secure_boot_available
+      ret = false
+      begin
+        ret = Yast::Execute.on_target(MOKUTIL, "--sb-state", allowed_exitstatus: 1) == 0
```

Comment:
I am not sure, but cannot be Systeminfo module used for checking secure boot availability?

Reply:
That has been my first thought too :-) . But in that state Systeminfo has not returned the correct value. So I have used mokutil. If it is worth I can check it again.....